src/us/kbase/shock/client/BasicShockClient.java

```diff
@@ -422,14 +427,17 @@ public ShockNode addNode(final InputStream file, final String filename, final St
 			throw new IllegalArgumentException(
 					"filename may not be null or empty");
 		}
+		if (fileLength < 0) {
+			throw new IllegalArgumentException("fileLength may not be negative");
```

Comment:
Is there a particular exception thrown for when the actual fileLength doesn't match the claimed fileLength? I didn't see one looking at the code, did I just miss it? The docs say the file will fail to save, but a descriptive error would be useful to developers.

Reply:
Shock doesn't care, but the blobstore will throw an error that says the content length is incorrect. That'll wind up as a `ShockHTTPException` here.

Reply:
I ran the TryShock code against the blobstore with a bad length and it was pretty clear what the problem was.